pkg/api/devops/v1alpha3/devopsproject_types.go

```diff
@@ -23,20 +23,121 @@ import (
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
-const DevOpsProjectFinalizerName = "devopsproject.finalizers.kubesphere.io"
-
 const (
 	ResourceKindDevOpsProject      = "DevOpsProject"
 	ResourceSingularDevOpsProject  = "devopsproject"
 	ResourcePluralDevOpsProject    = "devopsprojects"
 	DevOpsProjectPrefix            = "devopsproject.devops.kubesphere.io/"
+	DevOpsProjectFinalizerName     = "devopsproject.finalizers.kubesphere.io"
 	DevOpeProjectSyncStatusAnnoKey = DevOpsProjectPrefix + "syncstatus"
 	DevOpeProjectSyncTimeAnnoKey   = DevOpsProjectPrefix + "synctime"
-	DevOpeProjectSyncMsgAnnoKey    = DevOpsProjectPrefix + "syncmsg"
 )
 
 // DevOpsProjectSpec defines the desired state of DevOpsProject
 type DevOpsProjectSpec struct {
+	Argo *Argo `json:"argo,omitempty"`
```

Comment:
IMO, we should create a new version(like v1alpha4) of DevOpsProject instead of editing previous version(v1alpha3).

Reply:
It needs too much work if we have a new version.

Reply:
Of cause, I understand that it's not easy. But we should consider a scenario: upgrading from a previous version. If we do like this, the feature will not work as we want. Maybe we need to do more work to fix compatibility issues.

Reply:
> we should consider a scenario: upgrading from a previous version

There is no compatible issue for this PR. Because it only adds some new fields instead of modifying anything.